draft-ietf-ppm-dap.md

```diff
@@ -1186,27 +1271,157 @@ the last round is reached and each Aggregator computes an output share. All
+  is_leader: boolean indicating whether the host is the Leader
+  agg_param: aggregation parameter for the aggregation job
+  report_id: report ID
+  public_share: report public share
+  host_input_share: host's input share for the report
+
+def VdafPrepInit(vdaf_verify_key,
+                 is_leader,
+                 agg_param,
+                 report_id,
+                 public_share,
+                 host_input_share):
+  try:
+    host_prep_state = Vdaf.prep_init(
+      vdaf_verify_key,
+      0 if is_leader else 1,
+      agg_param,
+      report_id,
+      public_share,
+      host_input_share,
+    )
+    state CONTINUED(0, host_prep_state)
+  except:
+    state REJECTED
+~~~
+
+If the state is `REJECTED`, then processing halts. Otherwise, if the state is
+`CONTINUE`, then processing continues using the following transition rule until
+a terminal state is reached.
+
+~~~ transition
+State CONTINUED(round, host_prep_state)
+
+Input
+
+  is_leader: boolean indicating whether the host is the Leader
+  agg_param: aggregation parameter for the aggregation job
+  inbound: PrepareStep (or None)
+
+Output
+
+  outbound: PrepareStep (or None)
+
+def VdafPrepNext(is_leader, agg_param, inbound):
+  if type(inbound) == PrepareStep.initialize:
+    prep_msg, peer_prep_share = None, inbound.prep_share
+  elif type(inbound) = PrepareStep.continue:
+    prep_msg, peer_prep_share = inbound.prep_msg, inbound.prep_share
+  elif type(inbound) = PrepareStep.finish:
+    prep_msg, peer_prep_share = inbound.prep_msg, None
+  else:
+    prep_msg, peer_prep_share = None, None
+
+  try:
+    /* If prep_msg == None then this is the start of the
+     * first round. Otherwise, this is the prep message
+     * from the end of the previous round. */
+    out = Vdaf.prep_next(host_prep_state, prep_msg)
+    if round == Vdaf.ROUNDS:
+      state FINISHED(out)
+      /* The Leader reaches this state upon processing a response
+       * from the Helper, so there is no outbound message. */
+      return None if is_leader else PrepareStep.finished
+    (host_prep_state, host_prep_share) = out
+    round += 1
+
+    if peer_prep_share != None:
+      prep_shares = [peer_prep_share, host_prep_share]
+      if is_leader:
+        prep_shares.reverse()
+      prep_msg = Vdaf.prep_shares_to_prep(
+        agg_param,
+        prep_shares,
+      )
+      out = Vdaf.prep_next(host_prep_state, prep_msg)
+      if round == Vdaf.ROUNDS:
+        state FINISHED(out)
+        return PrepareStep.finish(prep_msg)
+      (host_prep_state, host_prep_share) = out
+
+    state CONTINUED(round, host_prep_state)
+    if is_leader and round == 1:
+      return PrepareStep.initialize(host_prep_share)
+    else:
+      return PrepareStep.continue(prep_msg, host_prep_share)
+  except:
+    state REJECT
+    return None
+~~~
+
+State `FINISHED` represents the state in which an output share (denoted
+`out_share`) was successfully recovered by the host.
+
```

Comment:
Reviewer note: I'm not sure if we want to keep the figure below. It was helpful for working out the state machine details, but it has a lot of redundant information (cf. {{agg-flow}}).

Reply:
I don't object to keeping this information around, but I think this ladder diagram should be merged with the one above (https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/pull/461/files#diff-8dbbcb0d121b441734a70cc867271480731a494b50eefdd758c249fdfb5ab84eR1107-R1138).

Reply:
Removed.